cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java

```diff
@@ -126,6 +132,78 @@ public CloseableIterator<StructuredRow> scan(Range keyRange, int limit) throws I
                              limit);
   }
 
+  @Override
+  public CloseableIterator<StructuredRow> scan(Range keyRange, int limit, String orderByField, SortOrder sortOrder)
+    throws InvalidFieldException, IOException {
+    LOG.trace("Table {}: Scan range {} with limit {}, sort field {} and sort order {}", schema.getTableId(),
+              keyRange, limit, orderByField, sortOrder);
+    if (!schema.isIndexColumn(orderByField)) {
+      throw new InvalidFieldException(schema.getTableId(), orderByField, "is not an indexed column");
+    }
+    ScannerIterator scannerIterator = null;
+    Comparator<StructuredRow> comparator = sortOrder.equals(SortOrder.ASC) ?
+      (row1, row2) ->
+        compareRowFields(row1, row2, orderByField)
+      : (row1, row2) ->
+      compareRowFields(row2, row1, orderByField);
+    List<StructuredRow> sorted = new ArrayList<>();
+
+    try {
+      scannerIterator = new ScannerIterator(getScanner(keyRange), schema);
+      while (scannerIterator.hasNext()) {
+        StructuredRow row = scannerIterator.next();
+        sorted.add(row);
+      }
+    } finally {
+      if (scannerIterator != null) {
+        scannerIterator.close();
+      }
+    }
+    Collections.sort(sorted, comparator);
```

Comment:
Perhaps add a comment to explain that we are doing the sorting in memory for nosql implementation only? For SQL we're using Database sortBy